--- a/Whirlpool.c	2003-03-13 14:46:38.000000000 +0600
+++ b/Whirlpool.c	2014-10-04 18:25:06.000000000 +0600
@@ -1824,7 +1824,7 @@
                 }
                 NESSIEfinalize(&w, computedDigest);
                 if (memcmp(computedDigest, expectedDigest, DIGESTBYTES) != 0) {
-                    fprintf(stderr, "API error @ pieceLen = %lu\n", pieceLen);
+                    fprintf(stderr, "API error @ pieceLen = %lu\n", (unsigned long)pieceLen);
                     display(computedDigest, DIGESTBYTES); printf("\n\n");
                     display(expectedDigest, DIGESTBYTES); printf("\n\n");
                     return;
@@ -1858,49 +1858,49 @@
 
     printf("2. In this example the data-string consists of a single byte, namely the ASCII-coded version of the letter 'a'.\n\n");
     NESSIEinit(&w);
-    NESSIEadd("a", 8*1, &w);
+    NESSIEadd((unsigned char *)"a", 8*1, &w);
     NESSIEfinalize(&w, digest);
     printf("The hash-code is the following 512-bit string.\n\n");
     display(digest, DIGESTBYTES); printf("\n\n");
 
     printf("3. In this example the data-string is the three-byte string consisting of the ASCII-coded version of 'abc'.\n\n");
     NESSIEinit(&w);
-    NESSIEadd("abc", 8*3, &w);
+    NESSIEadd((unsigned char *)"abc", 8*3, &w);
     NESSIEfinalize(&w, digest);
     printf("The hash-code is the following 512-bit string.\n\n");
     display(digest, DIGESTBYTES); printf("\n\n");
 
     printf("4. In this example the data-string is the 14-byte string consisting of the ASCII-coded version of 'message digest'.\n\n");
     NESSIEinit(&w);
-    NESSIEadd("message digest", 8*14, &w);
+    NESSIEadd((unsigned char *)"message digest", 8*14, &w);
     NESSIEfinalize(&w, digest);
     printf("The hash-code is the following 512-bit string.\n\n");
     display(digest, DIGESTBYTES); printf("\n\n");
 
     printf("5. In this example the data-string is the 26-byte string consisting of the ASCII-coded version of 'abcdefghijklmnopqrstuvwxyz'.\n\n");
     NESSIEinit(&w);
-    NESSIEadd("abcdefghijklmnopqrstuvwxyz", 8*26, &w);
+    NESSIEadd((unsigned char *)"abcdefghijklmnopqrstuvwxyz", 8*26, &w);
     NESSIEfinalize(&w, digest);
     printf("The hash-code is the following 512-bit string.\n\n");
     display(digest, DIGESTBYTES); printf("\n\n");
 
     printf("6. In this example the data-string is the 62-byte string consisting of the ASCII-coded version of 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'.\n\n");
     NESSIEinit(&w);
-    NESSIEadd("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789", 8*62, &w);
+    NESSIEadd((unsigned char *)"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789", 8*62, &w);
     NESSIEfinalize(&w, digest);
     printf("The hash-code is the following 512-bit string.\n\n");
     display(digest, DIGESTBYTES); printf("\n\n");
 
     printf("7. In this example the data-string is the 80-byte string consisting of the ASCII-coded version of eight repetitions of '1234567890'.\n\n");
     NESSIEinit(&w);
-    NESSIEadd("12345678901234567890123456789012345678901234567890123456789012345678901234567890", 8*80, &w);
+    NESSIEadd((unsigned char *)"12345678901234567890123456789012345678901234567890123456789012345678901234567890", 8*80, &w);
     NESSIEfinalize(&w, digest);
     printf("The hash-code is the following 512-bit string.\n\n");
     display(digest, DIGESTBYTES); printf("\n\n");
 
     printf("8. In this example the data-string is the 32-byte string consisting of the ASCII-coded version of 'abcdbcdecdefdefgefghfghighijhijk'.\n\n");
     NESSIEinit(&w);
-    NESSIEadd("abcdbcdecdefdefgefghfghighijhijk", 8*32, &w);
+    NESSIEadd((unsigned char *)"abcdbcdecdefdefgefghfghighijhijk", 8*32, &w);
     NESSIEfinalize(&w, digest);
     printf("The hash-code is the following 512-bit string.\n\n");
     display(digest, DIGESTBYTES); printf("\n\n");
@@ -1926,14 +1926,14 @@
 
     printf("3. In this example the data-string is the three-byte string consisting of the ASCII-coded version of 'abc'.\n\n");
     NESSIEinit(&w);
-    NESSIEadd("abc", 8*3, &w);
+    NESSIEadd((unsigned char *)"abc", 8*3, &w);
     NESSIEfinalize(&w, digest);
     printf("The hash-code is the following 512-bit string.\n\n");
     display(digest, DIGESTBYTES); printf("\n\n");
 
     printf("8. In this example the data-string is the 32-byte string consisting of the ASCII-coded version of 'abcdbcdecdefdefgefghfghighijhijk'.\n\n");
     NESSIEinit(&w);
-    NESSIEadd("abcdbcdecdefdefgefghfghighijhijk", 8*32, &w);
+    NESSIEadd((unsigned char *)"abcdbcdecdefdefgefghfghighijhijk", 8*32, &w);
     NESSIEfinalize(&w, digest);
     printf("The hash-code is the following 512-bit string.\n\n");
     display(digest, DIGESTBYTES); printf("\n\n");
